meta/AggregationCloud.php

```diff
@@ -0,0 +1,157 @@
+<?php
+namespace dokuwiki\plugin\struct\meta;
+class AggregationCloud {
+    /**
+     * @var string the page id of the page this is rendered to
+     */
+    protected $id;
+    /**
+     * @var string the Type of renderer used
+     */
+    protected $mode;
+    /**
+     * @var \Doku_Renderer the DokuWiki renderer used to create the output
+     */
+    protected $renderer;
+    /**
+     * @var SearchConfig the configured search - gives access to columns etc.
+     */
+    protected $searchConfig;
+    /**
+     * @var Column[] the list of columns to be displayed
+     */
+    protected $columns;
+    /**
+     * @var  Value[][] the search result
+     */
+    protected $result;
+    /**
+     * @var int number of all results
+     */
+    protected $resultCount;
+    /**
+     * @var string[] the result PIDs for each row
+     */
+    protected $resultPIDs;
+    /**
+     * @var array for summing up columns
+     */
+    protected $sums;
+    /**
+     * @var bool skip full table when no results found
+     */
+    protected $simplenone = true;
+    /**
+     * @todo we might be able to get rid of this helper and move this to SearchConfig
+     * @var \helper_plugin_struct_config
+     */
+    protected $helper;
+    /**
+     * Initialize the Aggregation renderer and executes the search
+     *
+     * You need to call @see render() on the resulting object.
+     *
+     * @param string $id
+     * @param string $mode
+     * @param \Doku_Renderer $renderer
+     * @param SearchConfig $searchConfig
+     */
+    public function __construct($id, $mode, \Doku_Renderer $renderer, SearchCloud $searchConfig) {
+        $this->id = $id;
+        $this->mode = $mode;
+        $this->renderer = $renderer;
+        $this->searchConfig = $searchConfig;
+        $this->data = $searchConfig->getConf();
+        $this->columns = $searchConfig->getColumns();
+        $this->result = $this->searchConfig->execute();
+        $this->resultCount = $this->searchConfig->getCount();
+        $this->helper = plugin_load('helper', 'struct_config');
+
+        $this->max = $this->result[0]['count'];
+        $this->min = end($this->result)['count'];
+    }
+
+    /**
+     * Create the table on the renderer
+     */
+    public function render() {
+        $this->startScope();
+        $this->renderer->doc .= '<ul>';
+        foreach ($this->result as $result) {
+            $this->renderResult($result);
+        }
+        $this->renderer->doc .= '</ul>';
+        $this->finishScope();
+        return;
+    }
+    /**
+     * Adds additional info to document and renderer in XHTML mode
+     *
+     * @see finishScope()
+     */
+    protected function startScope() {
+        // unique identifier for this aggregation
+        $this->renderer->info['struct_cloud_hash'] = md5(var_export($this->data, true));
+        // wrapping div
+        if($this->mode != 'xhtml') return;
+        $this->renderer->doc .= "<div class=\"structaggregation cloudaggregation\">";
```

Comment:
do we need both classes? do we reuse structaggregation styles? cloudaggregation would probably better named structcloud.